### PR TITLE
DOC: update/fix autoscaling documentation

### LINKED ISSUE #25613: [Doc]: better document default margins
I notice that the [autoscaling section in the user guide](https://matplotlib.org/devdocs/users/explain/axes/autoscale.html#margins) states
> The default margin around the data limits is 5%

This should perhaps be updated to indicate that it defaults to the rcParam (which defaults to 5%).  Possibly the docstrings of the various margin setting methods could also be updated to state where the default comes from.

_Originally posted by @rcomer in https://github.com/matplotlib/matplotlib/issues/25599#issuecomment-1493407846_

### PR DESCRIPTION
**Pull request by alissaahodge**
## This PR resolves #25613 by fixing the documentation in the margin section of the autoscaling documentation page so that it is easier to follow and understand. 
Before being opened as an issue, this comment was initially under another issue of the same topic. This can be found here: https://github.com/matplotlib/matplotlib/issues/25599#issuecomment-1493407846

## PR Checklist

**Documentation and Tests**
- [N/A ] Has pytest style unit tests (and `pytest` passes)
- [N/A ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ N/A] New plotting related features are documented with examples.

**Release Notes**

- [N/A ]  New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [N/A ]  API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
 - [N/A ] Release notes conform with instructions in `next_whats_new/README.rst or next_api_changes/README.rst`

### CODE REVIEW

File: galleries/users_explain/axes/autoscale.py [hunk 1]
```diff
@@ -29,12 +29,14 @@
 # %%
 # Margins
 # -------
-# The default margin around the data limits is 5%:
+# The default margin around the data limits is 5%, which is based on the default
+# configuration setting of rcParam:
```
Comment (QuLogic) [inline galleries/users_explain/axes/autoscale.py:33]: This should reference the specific rcParams:
```suggestion
# configuration setting of :rc:`axes.xmargin`, :rc:`axes.ymargin`, and :rc:`axes.zmargin`:
```

**Comment by QuLogic** [comment]
Please wrap the lines to ensure linting passes.

**Comment by melissawm** [comment]
Hello, @alissaahodge - the CircleCI failures here are unrelated (see #25672). Welcome!

**Comment by alissaahodge** [comment]
Thanks! cause I'm not really sure why there's been test failures on this PR when it codedev says "No Files covered by tests were changed" in this PR

**Review by QuLogic — APPROVED** [review]

**Comment by QuLogic** [comment]
We do generally prefer rebasing over merging, but as there are also a few extra commits here, I will just squash while merging to avoid re-running all CI again.

**Comment by QuLogic** [comment]
Thanks @alissaahodge! Congratulations on your first PR to Matplotlib :tada: We hope to hear from you again.

**Pull request merged**